### PR TITLE
Add a build step and produce ES6 files.

### PR DESCRIPTION
**Pull request by givanse**
This PRs does the following:

 - convert to TypeScript `pretender.js` => `src/index.ts`
   -  the rollup config outputs the files:
      - `pretender.js` - this is the IIFE version, it should be working the same as before
      - `pretender.es.js` - this one is new
 - replace jscs with eslint (jscs merged with eslint)
 - bump the supported node versions
 - started using Yarn
 - depends on https://github.com/pretenderjs/FakeXMLHttpRequest/pull/46

~I'm still testing this with an actual app, but I think its ready to get reviewed and get a conversation started.~ done with that

 - simple repo that showcases usage: https://github.com/givanse/use-pretender-as-a-module
 - tested with ember-cli-mirage (`yarn link pretender`)
 - tested it with a brand new glimmer app (0.13.0)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,21 +1,23 @@
 {
   "name": "pretender",
-  "version": "2.1.0",
+  "version": "3.1.0",
```
Comment (givanse) [inline package.json]: Took the liberty of bumping the version number since this is a fairly large change to the code base.

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+ *   host: 'www.yahoo.com:1234',
+ *   protocol: 'https:',
+ *   search: '?test=yes',
+ *   hash: '#abc',
+ *   href: 'https://www.yahoo.com:1234/mypage?test=yes#abc',
+ *   pathname: '/mypage',
+ *   fullpath: '/mypage?test=yes'
+ * }
+ */
+function parseURL(url: string) {
```
Comment (givanse) [inline src/index.ts]: TS is working

File: test/modules_test.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import PretenderES from '../src/pretender.es';
```
Comment (givanse) [inline test/modules_test.ts]: Pretender imported as a ES6 module. Also not that this test file is a `ts` file.

File: README.md [hunk 4]
```diff
@@ -40,6 +38,23 @@ const server = new Pretender(function() {
...
+```
+
+Or as a module.
+
+```javascript
+import Pretender from 'pretender';
+const server = new Pretender(function() {});
+```
+
+Full example: [use-pretender-as-a-module](https://github.com/givanse/use-pretender-as-a-module)
```
Comment (givanse) [inline README.md]: For the repo linked here, I added as collaborators the people that have merge access to Pretender.

Or if you prefer me to delete the mention of 'use-pretender-as-a-module', that is fine, I will.

**Comment by mike-north** [comment]
Is there any reason why all of these
> * convert to TypeScript pretender.js => src/index.ts
>    * the rollup config outputs the files:
>       * pretender.js - this is the IIFE version, it should be working the same as before
>       * pretender.es.js - this one is new
> * replace jscs with eslint (jscs merged with eslint)
> * bump the supported node versions
> * started using Yarn

must be included in a single PR? I could see this being at least 4

1. use yarn
2. jscs -> eslint
3. es module conversion (changing existing tests as little as possible, in order to prove that this is purely a packaging change)
4. ts conversion (it's nearly _always_ a better idea to do this as an independent PR)

This also has the potential to be a minor release.

**Comment by givanse** [comment]
you are right, this got too messy, I'll split it up

**Pull request closed without merging**